File: Arbitration/MPArbitration/Model/EMRClaimAttachment.cs
using System.Text.Json.Serialization;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace MPArbitration.Model
{
    public class BaseAttachment
    {
        [JsonPropertyName("id")]
        public int Id { get; set; } = 0;

        [JsonPropertyName("blobLink")]
        [StringLength(512)]
        public string BLOBLink { get; set; } = "";

        [Required]
        [JsonPropertyName("blobName")]
        [StringLength(255)]
        public string BLOBName { get; set; } = "";

        [JsonPropertyName("createdBy")]
        [StringLength(60)]
        public string CreatedBy { get; set; } = "";

        [JsonPropertyName("createdOn")]
        public DateTime? CreatedOn { get; set; } = null;

        [Required]
        [JsonPropertyName("docType")]
        public string DocType { get; set; } = string.Empty;

        [JsonPropertyName("isDeleted")]
        public bool IsDeleted { get; set; } = false;

        [JsonPropertyName("updatedBy")]
        [StringLength(60)]
        public string UpdatedBy { get; set; } = "";

        [JsonPropertyName("updatedOn")]
        public DateTime? UpdatedOn { get; set; } = null;
    }

    [Index(nameof(ArbitrationCaseId), nameof(DocType))]
    public class EMRClaimAttachment : BaseAttachment
    {
        [Required]
        [JsonPropertyName("arbitrationCaseId")]
        public int ArbitrationCaseId { get; set; } = 0;

    }

    [Index(nameof(AuthorityDisputeId), nameof(DocType), IsUnique = false)]
    public class AuthorityDisputeAttachment : BaseAttachment
    {
        [Required]
        [JsonPropertyName("authorityDisputeId")]
        public int AuthorityDisputeId { get; set; } = 0;

    }
}

File: Arbitration/MPArbitration/ClientApp/src/app/model/emr-claim-attachment.ts
export class BaseAttachment {
    public id = 0;
    public blobLink = '';
    public blobName = '';
    public createdBy = '';
    public createdOn:Date|undefined;
    public docType = '';
    public isDeleted = false;
    public updatedBy = '';
    public updatedOn:Date|undefined;

    constructor(obj?:any){
        if(!obj)
            return;

        Object.assign(this,JSON.parse(JSON.stringify(obj)));
        if(this.createdOn)
            this.createdOn = new Date(obj.createdOn);
    
        if(this.updatedOn)
            this.updatedOn = new Date(obj.updatedOn);

    }
}

export class AuthorityDisputeAttachment extends BaseAttachment{
    public authorityDisputeId = 0;
    constructor(obj?:any){
        super(obj);
        if(!obj)
            return;
        this.authorityDisputeId = obj.authorityDisputeId ?? 0;
    }
}

export class EMRClaimAttachment extends BaseAttachment{
    public arbitrationCaseId = 0;
    constructor(obj?:any){
        super(obj);
        if(!obj)
            return;
        this.arbitrationCaseId = obj.arbitrationCaseId ?? 0;
    }
}
